metrics/test/features/dashboard/domain/usecases/receive_build_metrics_updates_test.dart

```diff
@@ -0,0 +1,187 @@
+import 'package:metrics/features/dashboard/domain/repositories/metrics_repository.dart';
+import 'package:metrics/features/dashboard/domain/usecases/parameters/build_metrics_loading_param.dart';
+import 'package:metrics/features/dashboard/domain/usecases/receive_build_metrics_updates.dart';
+import 'package:metrics/util/date_time_util.dart';
+import 'package:test/test.dart';
+
+void main() {
+  group("Build metrics data loading", () {
+    const projectId = 'projectId';
+    const repository = MetricsRepositoryStubImpl();
+    final receiveBuildUpdates = ReceiveBuildMetricsUpdates(repository);
+    const buildLoadingLimit = 5;
+    const buildLoadingPeriod = Duration(days: 4);
+
+    List<Build> builds;
+    Build lastBuild;
+
+    BuildMetrics buildMetrics;
+
+    setUpAll(() async {
+      builds = MetricsRepositoryStubImpl.builds;
+
+      buildMetrics = await receiveBuildUpdates(const BuildMetricsLoadingParam(
+        projectId,
+        buildLoadingPeriod,
+        buildLoadingLimit,
+      )).first;
+
+      lastBuild = builds.last;
+    });
+
+    test('Calculates the average build time and total number of builds', () {
+      final expectedAverageBuildTime = builds
+              .map((build) => build.duration)
+              .reduce((value, element) => value + element) ~/
+          builds.length;
+      final expectedTotalBuildNumber = builds.length;
+
+      expect(buildMetrics.averageBuildTime, expectedAverageBuildTime);
+      expect(buildMetrics.totalBuildsNumber, expectedTotalBuildNumber);
+    });
+
+    test("Properly loads the performance metrics", () {
+      final firstPerformanceMetric = buildMetrics.performanceMetrics.first;
+
+      expect(
+        builds.length,
+        buildMetrics.performanceMetrics.length,
+      );
+
+      expect(
+        lastBuild.startedAt,
+        firstPerformanceMetric.date,
+      );
+      expect(
+        lastBuild.duration,
+        firstPerformanceMetric.duration,
+      );
+    });
+
+    test("Loads all fields in the build number metrics", () {
+      final buildStartDate = DateTimeUtil.trimToDay(lastBuild.startedAt);
+
+      final numberOfBuilds = builds
+          .where((element) =>
+              DateTimeUtil.trimToDay(element.startedAt) == buildStartDate)
+          .length;
+
+      final buildNumberMetrics = buildMetrics.buildNumberMetrics;
+      final firstBuildMetric = buildNumberMetrics.first;
+
+      expect(firstBuildMetric.date, buildStartDate);
+      expect(firstBuildMetric.numberOfBuilds, numberOfBuilds);
+    });
+
+    test('Properly loads the build result metrics', () {
+      final buildResultMetrics = buildMetrics.buildResultMetrics;
+
+      final firstBuildResultMetric = buildResultMetrics.first;
+
+      expect(firstBuildResultMetric.result, lastBuild.result);
+      expect(firstBuildResultMetric.duration, lastBuild.duration);
+      expect(firstBuildResultMetric.date, lastBuild.startedAt);
+      expect(firstBuildResultMetric.url, lastBuild.url);
```

Comment:
How about making the `BuildResultMetric` class `Equatable` to compare its instances easier? This will allow, for example: 
```dart
expect(firstBuildResultMetric, equals(lastBuild));
```